sqs/provider_test.go

```diff
@@ -0,0 +1,888 @@
+			BeforeEach(func() {
+				provisionData = provideriface.ProvisionData{
+					Details: domain.ProvisionDetails{
+						RawParameters: json.RawMessage(`{
+							"contentBasedDeduplication": false
+						}`),
+					},
+				}
+			})
+			It("should not set content-based-deduplication", func() {
+				Expect(queue.ContentBasedDeduplication).To(BeFalse())
+			})
+		})
+
+		Context("when contentBasedDeduplication provision param set to true", func() {
+			BeforeEach(func() {
+				provisionData = provideriface.ProvisionData{
+					Details: domain.ProvisionDetails{
+						RawParameters: json.RawMessage(`{
+							"contentBasedDeduplication": true
+						}`),
+					},
+				}
+			})
+
+			It("should set content-based-deduplication", func() {
+				Expect(queue.ContentBasedDeduplication).To(BeTrue())
+			})
+		})
+
+		Context("when delaySeconds provision param set", func() {
+			BeforeEach(func() {
+				provisionData = provideriface.ProvisionData{
+					Details: domain.ProvisionDetails{
+						RawParameters: json.RawMessage(`{
+							"delaySeconds": 5
+						}`),
+					},
+				}
+			})
+
+			It("should set queue delay in seconds", func() {
+				Expect(queue.DelaySeconds).To(Equal(5))
+			})
+		})
+
+		Context("when max message size provision param set", func() {
+			BeforeEach(func() {
+				provisionData = provideriface.ProvisionData{
+					Details: domain.ProvisionDetails{
+						RawParameters: json.RawMessage(`{
+							"maximumMessageSize": 10
+						}`),
+					},
+				}
+			})
+
+			It("should set the queue max message size", func() {
+				Expect(queue.MaximumMessageSize).To(Equal(10))
+			})
+		})
+
+		Context("when retention period provision param set", func() {
+			BeforeEach(func() {
+				provisionData = provideriface.ProvisionData{
+					Details: domain.ProvisionDetails{
+						RawParameters: json.RawMessage(`{
+							"messageRetentionPeriod": 3
+						}`),
+					},
+				}
+			})
+
+			It("should set the queue retention period", func() {
+				Expect(queue.MessageRetentionPeriod).To(Equal(3))
+			})
+		})
+
+		Context("when receive message wait time param set", func() {
+			BeforeEach(func() {
+				provisionData = provideriface.ProvisionData{
+					Details: domain.ProvisionDetails{
+						RawParameters: json.RawMessage(`{
+							"receiveMessageWaitTimeSeconds": 20
+						}`),
+					},
+				}
+			})
+
+			It("should set the wait time in seconds", func() {
+				Expect(queue.ReceiveMessageWaitTimeSeconds).To(Equal(20))
+			})
+		})
+
+		Context("when redrive max receive count param set", func() {
+			BeforeEach(func() {
+				provisionData = provideriface.ProvisionData{
+					Details: domain.ProvisionDetails{
+						RawParameters: json.RawMessage(`{
+							"redriveMaxReceiveCount": 30
+						}`),
+					},
+				}
+			})
+
+			It("should set redrive policy", func() {
+				Expect(queue.RedrivePolicy).To(And(
+					HaveKeyWithValue("maxReceiveCount", float64(30)),
+					HaveKey("deadLetterTargetArn"), // Is that good enough?
```

Comment:
feels like we left a TODO here without marking it as such

Reply:
ah yeah it's a TODO ... we can't test this with the way goformation works ... we'd just end up testing our own mock ... I'll remove the comment, the platform-tests could test this is want to extend them to test for things ending up on the queue (which maybe we should if we are gonna advertise that the DLQ is auto setup)